pkg/sif/write.go

```diff
@@ -94,8 +100,13 @@ func blobFromIndex(ii v1.ImageIndex, digest v1.Hash) (io.ReadCloser, error) {
 	return nil, errUnableToReadBlob
 }
 
-// writeIndexToFileImage writes an index and all of its child indexes, manifests and blobs to f.
-func (f *fileImage) writeIndexToFileImage(ii v1.ImageIndex, rootIndex bool) error {
+// writeIndex writes an index and all of its child indexes, manifests and blobs
+// to f. This function does not update the RootIndex.
+func (f *OCIFileImage) writeIndex(ii v1.ImageIndex) error {
+	return f.writeIndexorRootIndex(ii, false)
+}
+
+func (f *OCIFileImage) writeIndexorRootIndex(ii v1.ImageIndex, rootIndex bool) error {
```

Comment:
Could probably just keep this as one for now? Internal users can call with `rootIndex` set appropriately?

```suggestion
// writeIndex writes an index and all of its child indexes, manifests and blobs to f.
func (f *fileImage) writeIndex(ii v1.ImageIndex, rootIndex bool) error {
```